### ISSUE TITLE
[Tracking]: Move JOSS to use this framework

### ISSUE THREAD

**Issue by arfon** [openjournals/buffy#46]
Ultimately we want to move away from [Whedon-API](https://github.com/openjournals/whedon-api) to use this more generalised framework for JOSS.

This will require a number of changes under the hood:

- [x] Migrating the document production parts of Whedon (the background workers as well as the functionality in the [underlying gem](https://github.com/openjournals/whedon)) to use GitHub Actions and the Dockerfile.
- [x] Making sure we have coverage for all of the behaviours that Whedon API has today built here (checking lines of code, reminders, DOI verification etc.)
- [x] Moving the Whedon paper preview service to use the Dockerfile too.
- [ ] More things I'm sure...

Also, we'll need to verify the changes with JOSE and Julia Conference proceedings before we declare victory.

/ cc @xuanxu and @tarleb to expand on this list.

**Comment by jedbrown** [comment]
Is there a timeline on this? Inability to deploy changes to `latex.template` seems to be blocked on this issue, and that removes incentive to address issues such as https://github.com/openjournals/joss/issues/901.

**Comment by tarleb** [comment]
I don't think there's a detailed timeline other than "fairly soon", but any work on the LaTeX template and production can already happen in the new [Inara](https://github.com/openjournals/inara) project. This has the added benefit that we have a controlled environment, so shipping changes will be much easier.

**Comment by jedbrown** [comment]
Oh, thanks! Should I recreate https://github.com/openjournals/whedon/pull/105 there?

**Comment by tarleb** [comment]
@arfon may be able to move the issue from one repo to the other.

I hadn't seen the issue before, but I'll keep it in mind going forward!

**Comment by jedbrown** [comment]
It's a pull request to whedon's `latex.template` plus a font, and it's been in limbo because merging to whedon doesn't deploy, and thus was waiting for this issue. If Inara is ready to accept such PRs, I can create it there. I assume the Whedon PR should merge too, but am not familiar with how this sort of thing should be synchronized between repos.

**Comment by tarleb** [comment]
Ah sorry, I missed that it's a PR. A new PR would be welcome.
